docs/src/main/paradox/query.md

```diff
@@ -0,0 +1,136 @@
+
+Scala
+:  @@snip [create](/docs/src/test/scala/docs/home/query/QueryDocCompileOnly.scala) { #currentEventsByPersistenceId }
+
+### eventsBySlices
+
+The `eventsBySlices` and `currentEventsBySlices` queries are useful for retrieving all events for a given entity type.
+`eventsBySlices` should be used via Akka Projection.
+
+@@@ note
+
+This has historically been done with `eventsByTag` but the R2DBC plugin is instead providing `eventsBySlices`
+as an improved solution.
+
+The usage of `eventsByTag` for Projections has the drawback that the number of tags must be decided
+up-front and can't easily be changed afterwards. Starting with too many tags means much overhead since
+many projection instances would be running on each node in a small Akka Cluster. Each projection instance
+polling the database periodically. Starting with too few tags means that it can't be scaled later to more
+Akka nodes.
+
+With `eventsBySlices` more Projection instances can be added when needed and still reuse the offsets
+for the previous slice distributions.
+
+@@@
+
+A slice is deterministically defined based on the persistence id. The purpose is to evenly distribute all
+persistence ids over the slices. The `eventsBySlices` query is for a range of the slices. For example if
+using 128 slices and running 4 Projection instances the slice ranges would be 0-31, 32-63, 64-95, 96-127.
+Changing to 8 slice ranges means that the ranges would be 0-15, 16-31, 32-47, ..., 96-111, 112-127.
+
+Example of `currentEventsBySlices`:
+
+Java
+:  @@snip [create](/docs/src/test/java/jdocs/home/query/QueryDocCompileOnly.java) { #currentEventsBySlices }
+
+Scala
+:  @@snip [create](/docs/src/test/scala/docs/home/query/QueryDocCompileOnly.scala) { #currentEventsBySlices }
+
+`eventsBySlices` should be used via Akka Projection, which will automatically handle the following
+difficulties.
+
+The consumer can keep track of its current position in the event stream by storing the `offset` and restart the
+query from a given `offset` after a crash/restart.
+
+The offset is a `TimestampOffset` and it is based on the database `transaction_timestamp()` when the event was stored.
+`transaction_timestamp()` is the time when the transaction started, not when it was committed. This means that a
+"later" event may be visible first and when retrieving events after the previously seen timestamp may miss some
+events. In distributed SQL databases there can also be clock skews for the database timestamps. For that reason
+`eventsBySlices` will perform additional backtracking queries to catch missed events. Events from backtracking
+will typically be duplicates of previously emitted events. It's the responsibility of the consumer to filter
+duplicates and make sure that events are processed in exact sequence number order for each persistence id.
+Such deduplication is provided by the R2DBC Projection.
```

Comment:
What is maybe missing here is the ordering guarantees. 

The backtracking deliver missing events, but will we have events out of ordering? 
Yes, they can be out of ordering for the slice, but not for the same persistenceId. 

Maybe be good to clarify this.

Reply:
they can be out of order for a persistenceId because of the "missed events". I added more clarification of that in https://github.com/akka/akka-persistence-r2dbc/pull/149/commits/1fd44dfe7ecaccbb6cec7242c5053ad56b63d241

For example:
emitting 3, 4, 5
missing 6, 7
emitting 8, 9
backtracking emitting 5, 6, 7, 8, 9

meaning that a consumer will receive 8 before 6 and 7

Reply:
Hmm, I didn't know that it was possible to have out of order for the same persistence id. 
But isn't that so that the single write would make it unlikely, no?

Reply:
Out of order is a consequence of missed events, and missed events can happen when we move the offset forward (for another slice or pid). We can miss events when we switch to next "page" because we query from the latest offset.